### PR TITLE
dnsdist: Move constants to proper namespace

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
Redo of #5669. Separate PR because the rebase involved some manual work (some code got removed, moved around, etc. since), and this way we can compare the two PRs to make sure I did not miss anything.

~TODO: add the deprecation warnings as discussed in #5669.~ Pushed.

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by Habbie** [comment]
This is waiting for #7685, and I'd like to shuffle the commits around so that they prove that the -old- tests still pass under the -new- namespace setup.

File: pdns/dnsdistdist/docs/reference/constants.rst [hunk 1]
```diff
@@ -109,6 +118,22 @@ These constants represent an Action that can be returned from the functions invo
  * ``DNSAction.Truncate``: truncate the response
  * ``DNSAction.NoRecurse``: set rd=0 on the query
 
+.. _DNSQType:
+
+QType
+-----
+
+.. versionchanged:: 1.3.0
+  The prefix is changed from ``dnsdist.`` to ``QType``.
```
Comment (rgacogne) [inline pdns/dnsdistdist/docs/reference/constants.rst]:
```suggestion
  The prefix is changed from ``dnsdist.`` to ``DNSQType``.
```

File: pdns/dnsdistdist/docs/reference/constants.rst [hunk 2]
```diff
@@ -109,6 +118,22 @@ These constants represent an Action that can be returned from the functions invo
  * ``DNSAction.Truncate``: truncate the response
  * ``DNSAction.NoRecurse``: set rd=0 on the query
 
+.. _DNSQType:
+
+QType
+-----
```
Comment (rgacogne) [inline pdns/dnsdistdist/docs/reference/constants.rst]: Shouldn't that be `DNSQType` (and the number of `---` adjusted accordingly)?

**Comment by rgacogne** [comment]
#7685 has been merged, but no pressure ;-)

**Comment by Habbie** [comment]
> This is waiting for #7685, and I'd like to shuffle the commits around so that they prove that the -old- tests still pass under the -new- namespace setup.

I have now run the old tests under the new namespaces, and as long as I tolerate the deprecation warnings during startup in the test suite, all tests pass.

Ready for review, merge.

**Pull request merged**